src/appengine/libs/auth.py

```diff
@@ -54,6 +59,66 @@ def is_current_user_admin():
   return bool(key.get())
 
 
+@memoize.wrap(memoize.FifoInMemory(1))
+def _project_number_from_id(project_id):
+  """Get the project number from project ID."""
+  resource_manager = build('cloudresourcemanager', 'v1')
+  result = resource_manager.projects().get(projectId=project_id).execute()
+  return result['projectNumber']
```

Comment:
nit:maybe add
assert result and 'projectNumber' in result ?

Reply:
Added a check to raise an error.